nuxeo-justified-grid/nuxeo-justified-grid.html

```diff
@@ -287,14 +336,111 @@
       },
 
       _isSelected: function(item) {
-        return this.selectedItems.indexOf(item) > -1;
+        return this._isIndexSelected(item._view.index);
+      },
+
+      _isIndexSelected: function(index) {
+        return this.selectedItems.indexOf(this.items[index]) > -1;
+      },
+
+      /**
+       * Adds a list of items to the grid. The `items` are computed and transformed into rows.
+       * This is the ideal way to add new items to the grid (e.g. when loading more items with infinite scroll behavior).
+       */
+      _addItems: function (newItems) {
+        // recompute last row before adding new items
+        if (this.rows.length > 0) {
+          // get last row items
+          var lastRowItems = this.rows[this.rows.length - 1].map(function(item) {
+            return this.items[item._view.index];
+          }.bind(this));
+          // append last row items to `newItems` param
+          newItems = lastRowItems.concat(newItems);
+          // drop last row
+          this.pop('rows');
+        }
+        this._computeRows(newItems).forEach(function (row) {
+          this.push('rows', row);
+        }.bind(this));
+        this.$.scrollThreshold.clearTriggers();
+      },
+
+      /**
+       * Given a set of items, computes a list of rows. The goal is to compute each row by trying to add items until
+       * they fit. When a row doesn't have space to fit another item, we increase the height of the current row to take
+       * advantage of the remaining width space to avoid cropping images.
+       * Currently relies on `picture:info` to get the dimensions of a video/image. Fallbacks to square dimensions if the
+       * document doesn't have the `picture` schema.
+       */
+      _computeRows: function(items) {
+        var gridWidth = this.$.list.offsetWidth;
+        var rows = new Array();
+        var currentRowWidth = 0;
+        var currentRow = new Array();
+        items.forEach(function(item, idx) {
+          var clone = Object.assign({}, item);
```

Comment:
`Object.assign` is not supported by IE11. Either we add a polyfill to [es6-polyfills.html](https://github.com/nuxeo/nuxeo-ui-elements/blob/master/es6-polyfills.html) or we use a loop like we're doing on [nuxeo-layout](https://github.com/nuxeo/nuxeo-ui-elements/blob/master/nuxeo-layout.html#L154-L159).